### PR TITLE
Feature/sc 32782/topics module route updates for clark refactor

### PR DESCRIPTION
**Pull request by Romerico234**
Removed and updated UPDATE_TOPIC route to point to topics.service.ts. DELETE_TOPIC route is not referenced anywhere.

### CODE REVIEW

**Review by Greek — CHANGES_REQUESTED** [review]
some file are calling the moved functions, see the trace here:

```
Error: src/app/admin/components/learning-object-list-item/learning-object-list-item.component.ts:252:27 - error TS2339: Property 'deleteRevision' does not exist on type 'UnreleaseService'.

252     this.unreleaseService.deleteRevision(this.learningObject.author.username, this.learningObject.cuid, this.learningObject.version + 1)
                              ~~~~~~~~~~~~~~


Error: src/app/cube/details/components/action-panel/editorial-action-pad/editorial-action-pad.component.ts:86:8 - error TS2339: Property 'createRevision' does not exist on type 'LearningObjectService'.

86       .createRevision(this.learningObject.cuid).then(async (revisionUri: any) => {
          ~~~~~~~~~~~~~~


Error: src/app/onion/dashboard/dashboard.component.ts:356:64 - error TS2339: Property 'createRevision' does not exist on type 'LearningObjectService'.

356     this.sidePanelPromiseResolver = this.learningObjectService.createRevision(object.cuid).then(() => {
                                                                   ~~~~~~~~~~~~~~


Error: src/app/onion/learning-object-builder/builder-store.service.ts:256:55 - error TS2339: Property 'createRevision' does not exist on type 'LearningObjectService'.

256         revisionId = await this.learningObjectService.createRevision(cuid);
                                                          ~~~~~~~~~~~~~~


Error: src/app/onion/relevancy-builder/builder-store.service.ts:140:35 - error TS2339: Property 'updateObjectTopics' does not exist on type 'RelevancyService'.

140       await this.relevancyService.updateObjectTopics(this._learningObject.id, this._topics);
                                      ~~~~~~~~~~~~~~~~~~

```

**Review by Cwagne17 — COMMENTED** [review]
Same thing for this PR, rebase then re-request.

File: src/app/core/learning-object-module/topics/topics.service.ts [hunk 1]
```diff
@@ -45,6 +45,36 @@ export class TopicsService {
...
+
+          catchError(this.handleError)
+        )
+        .toPromise()
+        .then(
+          (res: any) => resolve(res),
+          (err) => reject(err),
+        );
+    });
+  }
```
Comment (pzalep1) [inline src/app/core/learning-object-module/topics/topics.service.ts:76]: Make sure all the references to the old function get updated to this function.

Comment (stef359) [reply src/app/core/learning-object-module/topics/topics.service.ts:76]: ^

Comment (Cwagne17) [reply src/app/core/learning-object-module/topics/topics.service.ts:76]: If you run `npx ng serve` or `ng serve` if you have angular installed. We should be testing the changes we make prior to merging/PR reviews.

**Comment by Romerico234** [comment]
Removed test comment. Removed and updated UPDATE_TOPICS route. Fixed function calls. Renamed function parameter.

**Review by stef359 — APPROVED** [review]

**Review by Cwagne17 — COMMENTED** [review]
You mention that DELETE_TOPIC is not used anywhere but it isn't removed in the PR. Are we keeping this?

**Comment by Romerico234** [comment]
> You mention that DELETE_TOPIC is not used anywhere but it isn't removed in the PR. Are we keeping this?

I removed the DELETE_TOPIC route and I found no issues after its removal.

**Review by pzalep1 — APPROVED** [review]

**Review by dsoto18 — APPROVED** [review]
Looks good and makes sense to me.

**Review by Cwagne17 — APPROVED** [review]

**Review by johngonz0 — APPROVED** [review]

**Review by Greek — APPROVED** [review]

**Pull request merged**